examples_tests/common/CommonAPI.h

```diff
@@ -0,0 +1,287 @@
+		assert(result.commandPool);
+		result.physicalDevice = std::move(gpu);
+
+
+		return result;
+
+	}
+	static nbl::core::smart_refctd_ptr<nbl::video::ISwapchain> createSwapchain(uint32_t width,
+		uint32_t height,
+		uint32_t imageCount,
+		const nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice>& device,
+		const nbl::core::smart_refctd_ptr<nbl::video::ISurface>& surface,
+		nbl::video::ISurface::E_PRESENT_MODE presentMode)
+	{
+		using namespace nbl;
+		video::ISwapchain::SCreationParams sc_params;
+		sc_params.width = width;
+		sc_params.height = height;
+		sc_params.arrayLayers = 1u;
+		sc_params.minImageCount = imageCount;
+		sc_params.presentMode = presentMode;
+		sc_params.surface = surface;
+		sc_params.surfaceFormat.format = asset::EF_R8G8B8A8_SRGB;
+		sc_params.surfaceFormat.colorSpace.eotf = asset::EOTF_sRGB;
+		sc_params.surfaceFormat.colorSpace.primary = asset::ECP_SRGB;
+
+		return device->createSwapchain(std::move(sc_params));
+	}
+	static nbl::core::smart_refctd_ptr<nbl::video::IGPURenderpass> createRenderpass(const nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice>& device)
+	{
+		using namespace nbl;
+
+		video::IGPURenderpass::SCreationParams::SAttachmentDescription a;
+		a.initialLayout = asset::EIL_UNDEFINED;
+		a.finalLayout = asset::EIL_UNDEFINED;
+		a.format = asset::EF_R8G8B8A8_SRGB;
+		a.samples = asset::IImage::ESCF_1_BIT;
+		a.loadOp = video::IGPURenderpass::ELO_CLEAR;
+		a.storeOp = video::IGPURenderpass::ESO_STORE;
+
+		video::IGPURenderpass::SCreationParams::SSubpassDescription::SAttachmentRef colorAttRef;
+		colorAttRef.attachment = 0u;
+		colorAttRef.layout = asset::EIL_UNDEFINED;
+		video::IGPURenderpass::SCreationParams::SSubpassDescription sp;
+		sp.colorAttachmentCount = 1u;
+		sp.colorAttachments = &colorAttRef;
+		sp.depthStencilAttachment = nullptr;
+		sp.flags = video::IGPURenderpass::ESDF_NONE;
+		sp.inputAttachmentCount = 0u;
+		sp.inputAttachments = nullptr;
+		sp.preserveAttachmentCount = 0u;
+		sp.preserveAttachments = nullptr;
+		sp.resolveAttachments = nullptr;
+
+		video::IGPURenderpass::SCreationParams rp_params;
+		rp_params.attachmentCount = 1u;
+		rp_params.attachments = &a;
+		rp_params.dependencies = nullptr;
+		rp_params.dependencyCount = 0u;
+		rp_params.subpasses = &sp;
+		rp_params.subpassCount = 1u;
+
+		return device->createGPURenderpass(rp_params);
+	}
+
+	template<size_t imageCount, size_t width, size_t height>
+	static auto createFBOWithSwapchainImages(const nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice>& device,
+		nbl::core::smart_refctd_ptr<nbl::video::ISwapchain> swapchain,
+		nbl::core::smart_refctd_ptr<nbl::video::IGPURenderpass> renderpass)->std::array<nbl::core::smart_refctd_ptr<nbl::video::IGPUFramebuffer>, imageCount>
+	{
+		using namespace nbl;
+		std::array<nbl::core::smart_refctd_ptr<video::IGPUFramebuffer>, imageCount> fbo;
+		auto sc_images = swapchain->getImages();
+		assert(sc_images.size() == imageCount);
```

Comment:
just dont have the `imageCount` template param, `sc_images.size()` gives you the size, you can return `smart_refctd_dynamic_array` instead of a `std::array` from the function